### ISSUE TITLE
atilde (U+00E3) doesn't work as phrase while other unicode characters work

### ISSUE THREAD

**Issue by pserey** [autokey/autokey#579]
I've changed to Manjaro Linux a while ago but I can't make autokey work as it worked in Ubuntu. I use alt+a, alt+e and alt+o to type ã, ê and õ respectively with autokey phrases. I've setted the same phrases in my autokey-qt installation but alt+a simply doesn't work, while alt+e and alt+o are working as expected. Any fixes on this? 

![2021-07-15-094745_387x103_scrot](https://user-images.githubusercontent.com/65696885/125790665-8b877a73-f6e2-42bc-badf-8fa7c266cd96.png)
![2021-07-15-094809_382x99_scrot](https://user-images.githubusercontent.com/65696885/125790666-6386cc94-34f2-43ec-ab61-112794b2908b.png)

OS: Manjaro Linux
WM: Awesomwm

**Comment by josephj11** [comment]
Take a look at [this](https://github.com/autokey/autokey/wiki/Adding-Unicode-Characters-or-Emojis-to-Your-Scripts-or-Phrases) and see if it helps.

**Comment by pserey** [comment]
Hello! Already tried this, it weirdly returns the character code string look

Created a script with `keyboard.send_keys("<ctrl>+<shift>+u+00E3 ")` and binded it to `alt + a`, in terminal that's what i got:
![2021-07-16-210621_212x44_scrot](https://user-images.githubusercontent.com/65696885/126019155-9a41546b-c077-4d42-ac03-2ddb652b85a5.png)

It's really strange and I don't know if it may be any problem with the distro or window manager, because when I used ubuntu it worked like a charm.

PS: if the script has `keyboard.send_keys("<ctrl>+<shift>+u+00E3+")` it just adds the literal "+" in the output


PPS Tried making a phrase with `<ctrl>+<shift>+u+00E3 ` too and the output is the same

**Comment by josephj11** [comment]
That should have worked.

Did you try using the clipboard API method in a script? That almost always works with Unicode as long as your text doesn't include any active keys or macros (like `<left>` or `<cursor>`.

You add the actual Unicode character(s) to the string you're going to paste without using all the U codes.

You didn't specify which front end you're using. I assume it's autokey-gtk.

It would be interesting to see if you get the same results with the other front end (autokey-qt). It's a little tricky to install both front ends on Arch and derivatives because of the way their packaging system works, so if that's too much trouble for you, don't bother.

**Comment by pserey** [comment]
Oh, sorry! I'm using autokey-qt. I'll try that clipboard API method to see if it works, I assume it works like this `keyboard.send_keys("ã")`, right? If it doesnt work I'll try installing autokey-gtk to see if it behaves differently.

Edit: If this wasn't the right way to do it I'm sorry, but it didn't work. Now when I press the hotkey it outputs "à" instead of "ã"

**Comment by josephj11** [comment]
Nope, that's not pasting. To paste, you build your string somewhere - either in a variable or in the *clipboard* API call itself and then actually paste it. I don't know how your example above worked at all. keyboard.send_keys() can't deal with Unicode. I would have expected it to put out a space instead of the wrong Unicode character.

    clipboard.fill_clipboard("ã")
    time.sleep(0.1)
    keyboard.send_keys("<ctrl>+v")

is a simple example.

When you operate on the clipboard, it usually requires adding a delay after the API call to give it time to complete because the call returns successfully immediately even though the action is being processed asynchronously in another thread. That's an artifact of the way the upstream clipboard libraries we use work and applies to both front ends.

**Comment by pserey** [comment]
Ohhhh, that worked like a charm!! Thank you very much! I don't know why the phrases were not working, but I'll start using this as it seems more universal for every machine I use. I just changed the delay as 0.1 is a little to long for me as a typist haha.

Best regards!

**Comment by Elliria** [comment]
I'm glad to see that Joe has found a work-around that gets the job done by literally copying and pasting Unicode characters. I've 
[added that](https://github.com/autokey/autokey/wiki/Unicode-Characters#to-print-a-character-that-wont-print) to the [Unicode Characters](https://github.com/autokey/autokey/wiki/Unicode-Characters) wiki page.

Meanwhile, from what I've found in all of my experimentation, the technique used above for printing Unicode characters will usually only work on Unicode character-codes that use the base-ten positional numeral system and _don't contain letters_. There have been occasional exceptions, though, so it's worth it to try to see which ones will print and which won't. If it's got letters in it, give it the suspicious eye and go from there.

**Comment by josephj11** [comment]
Even though a suitable workaround has been found, it seems to me that this is still a bug somewhere - even if it's upstream from AutoKey. I'm going to reopen it so a developer can take a look at it.

**Comment by pserey** [comment]
I agree, it seems off that some unicode characters simply don't work in phrases.

**Comment by Elliria** [comment]
That sounds good. I'm not sure of its cause, but whoever fixes it will be a hero. This topic comes up often. We do all seem to like our decorative characters.
